src/components/datepicker/common/datepicker-button.ts

```diff
@@ -0,0 +1,201 @@
+      }
+    }
+  }
+
+  private _init(picker?: string | SbbDatepickerElement): void {
+    this._datePickerController?.abort();
+    this._datePickerController = new AbortController();
+    this.datePickerElement = getDatePicker(this, picker);
+    this.setDisabledState(this.datePickerElement);
+    if (!this.datePickerElement) {
+      // If the component is attached to the DOM before the datepicker, it has to listen for the datepicker init,
+      // assuming that the two components share the same parent element.
+      this.parentElement?.addEventListener(
+        'inputUpdated',
+        (e: CustomEvent<InputUpdateEvent>) => this._init(e.target as SbbDatepickerElement),
+        { once: true, signal: this._datePickerController.signal },
+      );
+      return;
+    }
+    this._setAriaLabel();
+
+    this.datePickerElement.addEventListener(
+      'change',
+      (event: Event) => {
+        this.setDisabledState(event.target as SbbDatepickerElement);
+        this._setAriaLabel();
+      },
+      { signal: this._datePickerController.signal },
+    );
+    this.datePickerElement.addEventListener(
+      'datePickerUpdated',
+      (event: Event) => {
+        this.setDisabledState(event.target as SbbDatepickerElement);
+        this._setAriaLabel();
+      },
+      { signal: this._datePickerController.signal },
+    );
+    this.datePickerElement.addEventListener(
+      'inputUpdated',
+      (event: CustomEvent<InputUpdateEvent>) => {
+        this._inputDisabled = !!(event.detail.disabled || event.detail.readonly);
+        this._setAriaLabel();
+        this.onInputUpdated(event);
+      },
+      { signal: this._datePickerController.signal },
+    );
+
+    this.datePickerElement.dispatchEvent(datepickerControlRegisteredEventFactory());
+  }
+
+  private _setAriaLabel(): void {
+    const currentDate = this.datePickerElement?.getValueAsDate?.();
+
+    if (!currentDate || !this._dateAdapter.isValid(currentDate)) {
+      this.setAttribute('aria-label', this.i18nOffBoundaryDay[this._language.current]);
+      return;
+    }
+
+    const currentDateString =
+      this.datePickerElement?.now().toDateString() === currentDate.toDateString()
+        ? i18nToday[this._language.current].toLowerCase()
+        : this._dateAdapter.getAccessibilityFormatDate(currentDate);
+
+    this.setAttribute(
+      'aria-label',
+      this.i18nSelectOffBoundaryDay(currentDateString)[this._language.current],
+    );
+  }
+
+  private _setDisabledRenderAttributes(): void {
+    this.toggleAttribute('data-disabled', this._disabled || this._inputDisabled);
+    if (isValidAttribute(this, 'data-disabled')) {
+      this.setAttribute('aria-disabled', 'true');
+      this.removeAttribute('tabindex');
+    } else {
+      this.removeAttribute('aria-disabled');
+      this.setAttribute('tabindex', '0');
+    }
+  }
+
+  protected override renderTemplate(): TemplateResult {
+    this._setDisabledRenderAttributes();
```

Comment:
Maybe use the moment and set the attributes when needed and not in every rendering?

Reply:
Do you feel should i move the method in the `willUpdate` callback?

Reply:
I removed it and called the method when the two variables are changed